Bottom: 3c3f0f8a447e86b0bbd6e61d3ab5e43a37ac7ce0
Top:    94a04020cb38c5549a09c4486652949bcfcdb733
Author: Mark de Wever <koraq@xs4all.nl>
Date:   2021-08-14 15:14:02 +0200

Refresh of add-a-calculator-value-object

---

diff --git a/src/modules/calculator/CMakeLists.txt b/src/modules/calculator/CMakeLists.txt
index 16ce5a7..c28dd37 100644
--- a/src/modules/calculator/CMakeLists.txt
+++ b/src/modules/calculator/CMakeLists.txt
@@ -7,25 +7,5 @@ target_compile_options(calculator
 	PRIVATE
 		${module_compile_options}
 )
-target_link_libraries(calculator
-	PRIVATE
-		math
-)
-add_module(calculator calculator.cpp)
-add_module(calculator.controller controller.cpp)
-add_module(calculator.model model.cpp)
 add_module(calculator.value value.cpp)
-add_dependencies(calculator.model.pcm
-	calculator.value.pcm
-)
-add_dependencies(calculator.controller.pcm
-	calculator.model.pcm
-)
-add_dependencies(calculator.pcm
-	calculator.value.pcm
-	calculator.model.pcm
-	calculator.controller.pcm
-)
-add_dependencies(calculator
-	calculator.pcm
-)
+add_dependencies(calculator calculator.value.pcm)
diff --git a/src/modules/calculator/value.cpp b/src/modules/calculator/value.cpp
index 667e40b..0411824 100644
--- a/src/modules/calculator/value.cpp
+++ b/src/modules/calculator/value.cpp
@@ -45,9 +45,6 @@ public:
 
 private:
   int64_t value_{0};
-
-  /** Only equality comparable is part of the contract. */
-  friend auto operator<=>(const tvalue &, const tvalue &) = default;
 };
 
 } // namespace calculator
